### PR TITLE
FluxC: Delete site integration

### LINKED ISSUE #5006: Delete Site FluxC integration
Delete site network call added to FluxC in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/266, we need to integrate that in.

### PR DESCRIPTION
**Pull request by oguzkocer**
Closes #5006. This PR implements the delete site flow that's introduced to FluxC in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/266. Currently the only issue is that once a site is deleted the Site Picker doesn't refresh and still shows the deleted site in the list. I opened a separate issue for that, since it'll most likely happen for removed sites as well #5018.

The deleted site should be removed from DB [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java#L682), but due to the issue explained above, I am still not sure if that's the case or not.

To test:
* Create a new WordPress.com site
* Go into the settings of that site
* Delete the site
* You should be taken back to the Site Picker page

### CODE REVIEW

**Comment by maxme** [comment]
Just tested: the site is deleted, but it's not removed from the DB.

**Comment by maxme** [comment]
I wonder if it's related to this: https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/281

**Comment by oguzkocer** [comment]
It could be. But if that was the case wouldn't `isError` always return `false` and the site always got removed from DB even if there was an error: https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java#L703-L705

**Comment by maxme** [comment]
I just tried in the FluxC's example app and the site is removed from the DB, so I don't think this is our problem.

**Comment by maxme** [comment]
I think I found the culprit: we're saving the site [here](https://github.com/wordpress-mobile/WordPress-Android/blob/935d936c3f4975b9a8d614458db1e8ba3dcb5f60/WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java#L249) so when the delete action ends, we're creating a new entry with the previous in memory instance.

**Comment by maxme** [comment]
@oguzkocer Fixed in 29191a0 - can you try it

**Comment by oguzkocer** [comment]
@maxme That fix is working for me, thanks for taking care of it!

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/BlogPreferencesActivity.java [hunk 1]
```diff
@@ -175,9 +175,21 @@ public void onEventMainThread(ConnectionChangeReceiver.ConnectionChangeEvent eve
 
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
-    public void onSiteRemoved(OnSiteRemoved event) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/BlogPreferencesActivity.java]: I think we still need that for self hosted sites.

Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/BlogPreferencesActivity.java]: Fixed in e576dbb3ad9efcf1304c41d9910a7928b74d73ed

**Comment by maxme** [comment]
:shipit:

**Pull request merged**